CMakeLists.txt

```diff
@@ -187,31 +187,36 @@ if(WITH_BLUESTORE)
 endif()
 
 include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(WITH_LIBBLK "Enable backend driver to access storage device" ON
```

Comment:
this is confusing, IMHO. what is backend driver? backend of what? and what it is supposed to drive?

Reply:
The right meaning is "the software stack" which will be used by bluestore or other components to use their interface to access to storage devices, such as SSD or AEP.
Please help update it if you have better idea about the narration.

Reply:
i cannot think of a name for this. in the context of a storage system talking to hard disk and other persistent storage , i think it's a must "to access storage device".  what does this imply if this option is disabled?

Reply:
@tchaikov 
When WITH_BLUESTORE or WITH_RBD_RWL is enabled,  WITH_LIBBLK should be ON.
WITH_BLUESTORE is default "ON", developers don't need to specify "WITH_LIBBLK" options unless they know what they really want to do(In this case, developers should already have the ability to check the build system)

Reply:
this option's doc reads:

> Enable backend driver to access storage device

if i were an end user, and i am trying to think like an end user now, i will wonder "why"? does this option disable ceph from working, if it's disabled?

> @tchaikov
> When WITH_BLUESTORE or WITH_RBD_RWL is enabled, WITH_LIBBLK should be ON.

this is not true, "WITH_BLUESTORE or WITH_RBD_RWL" is a necessary condition for enabling `WITH_LIBBLK`. it's not `WITH_LIBBLK` 's sufficient condition.

WITH_BLUESTORE OR WITH_RBD_RWL

> WITH_BLUESTORE is default "ON", developers don't need to specify "WITH_LIBBLK" options unless they know what they really want to do(In this case, developers should already have the ability to check the build system)

i can now deduce that the internal library of libblk is a dependent of both bluestore and rbd-rwl, but IMHO, this is not a reason to make it an option. libceph-common and libglobal are not exposed an optional component controlled by a user-visible option, even if they are used by lots of component in Ceph. i use they as an example, not because it's correct because it's how ceph is programmed this way. but because i don't think a user-visible option should help user to make a decision based on its document.

Reply:
@tchaikov 
I changed it to be:
```
option(WITH_LIBBLK "Backend driver to access storage device" ON)
```
Bluestore must use LIBBLK

Reply:
Remove WITH_LIBBLK option.